source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ARCGridViewLayout.mm

```diff
@@ -0,0 +1,369 @@
+        
+        NSMutableArray<UIView *> *columnViews = [NSMutableArray array];
+        for (NSInteger i = 0; i < _numberOfColumns; i++) {
+            UIView *columnView = [[UIView alloc] init];
+            columnView.translatesAutoresizingMaskIntoConstraints = NO;
+            [rowView addSubview:columnView];
+            NSString *type = _columnWidthTypes[i];
+            NSNumber *value = _columnWidthValues[i];
+            NSInteger previousAutoIndex = -1;
+            NSLayoutConstraint *widthConstraint;
+            if ([type isEqualToString:@"fixed"]) {
+                CGFloat columnWidth = [value floatValue];
+                widthConstraint = [columnView.widthAnchor constraintEqualToConstant:columnWidth];
+            } else if ([type isEqualToString:@"percentage"]) {
+                CGFloat percentage = [value floatValue] / 100.0;
+                widthConstraint = [columnView.widthAnchor constraintEqualToAnchor:self.widthAnchor multiplier:percentage];
+            } else if ([type isEqualToString:@"auto"]) {
+                if (previousAutoIndex != -1) {
+                    widthConstraint = [columnViews[previousAutoIndex].widthAnchor constraintEqualToAnchor:(columnView.widthAnchor)];
+                } else {
+                    widthConstraint = [columnView.widthAnchor constraintGreaterThanOrEqualToConstant:0];
+                }
+            }
+            
+            [NSLayoutConstraint activateConstraints:@[
+                [columnView.leadingAnchor constraintEqualToAnchor:(i == 0 ? rowView.leadingAnchor : columnViews[i - 1].trailingAnchor) constant:_columnSpacing],
+                [columnView.topAnchor constraintEqualToAnchor:rowView.topAnchor],
+                [columnView.bottomAnchor constraintEqualToAnchor:rowView.bottomAnchor],
+                widthConstraint
+            ]];
+            if (i == _numberOfColumns-1) {
+                [NSLayoutConstraint activateConstraints:@[
+                    [columnView.trailingAnchor constraintEqualToAnchor:rowView.trailingAnchor constant:-_columnSpacing]
+                ]];
+            }
+            
+            [columnViews addObject:columnView];
+        }
+        
+        if (rowIndex == 0) {
+            [NSLayoutConstraint activateConstraints:@[
+                [rowView.topAnchor constraintEqualToAnchor:self.topAnchor constant:_rowSpacing], // Add Top constraint
+                [rowView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+                [rowView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor]
+            ]];
+        } else {
+            UIView *previousRow = _rows.lastObject;
+            [NSLayoutConstraint activateConstraints:@[
+                [rowView.topAnchor constraintEqualToAnchor:previousRow.bottomAnchor constant:_rowSpacing],
+                [rowView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+                [rowView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor]
```

Comment:
can we follow objective c code style - move open braces to new line, else statements on new line